client/src/components/layout/TripShow.js

```diff
@@ -32,6 +34,50 @@ const TripShow = (props) => {
     showTrip()
   }, [])
 
+  const handleDeleteShowPage = async () => {
+    try {
+      const response = await fetch(`/api/v1/trips/${tripId}`, {
+        method: "DELETE",
+        headers: new Headers({
+          "Content-Type": "application/json",
+        }),
+      })
+      if (!response.ok) {
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
+      }
+      setShouldRedirect(true)
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  if (shouldRedirect === true) {
+    return <Redirect to="/" />
+  }
+
+  const handleEditShowPage = (event) => {
+    try {
+      setEditRedirect(true)
+    } catch (error) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
```

Comment:
haha I think you've gotten too used to fetch. delete this function. On line 75, replace your button with a <Link> tag from react router. That is how you should send folks to different pages.  Redirect should only be used if we are triggering a page change that isnt immediately reliant on a user click.
Further....there is no fetch request here :P